File: src/main/java/com/j/ming/socket/model/TransLocalFile.java
package com.j.ming.socket.model;



/**
 * Created by Sunny on 2017/5/13 0013.
 */

public class TransLocalFile {
    private String md5;
    private final String name;
    private final long size;
    private String path = "";

    private long createTime;
    private boolean done = false;
    private int progress = 0;
    private String rate;
    private int fileTAG = -1;
    public static final int TAG_SEND = 0;
    public static final int TAG_RECEIVE = 1;

    private TransLocalFile(Builder builder) {
        setMd5(builder.md5);
        name = builder.name;
        size = builder.size;
        setPath(builder.path);
        setCreateTime(builder.createTime);
        setDone(builder.done);
        setProgress(builder.progress);
        setRate(builder.rate);
        fileTAG = builder.fileTAG;
    }


    public String getMd5() {
        return md5;
    }

    public void setPath(String path) {
        this.path = path;
    }

    public void setMd5(String md5) {
        this.md5 = md5;
    }

    public void setDone(boolean done) {
        this.done = done;
    }

    public void setCreateTime(long createTime) {
        this.createTime = createTime;
    }

    public void setProgress(int progress) {
        this.progress = progress;
    }

    public void setRate(String rate) {
        this.rate = rate;
    }

    public String getName() {
        return name;
    }

    public long getSize() {
        return size;
    }

    public long getCreateTime() {
        return createTime;
    }

    public boolean isDone() {
        return done;
    }

    public String getPath() {
        return path;
    }

    public int getProgress() {
        return progress;
    }

    public String getRate() {
        return rate;
    }

    public int getFileTAG() {
        return fileTAG;
    }

    public void setFileTAG(int fileTAG) {
        this.fileTAG = fileTAG;
    }

    public static final class Builder {
        private String name;
        private long size;
        private long createTime;
        private boolean done;
        private String path;
        private int progress;
        private String rate;
        private int fileTAG;
        private String md5;

        public Builder() {
        }

        public Builder name(String val) {
            name = val;
            return this;
        }

        public Builder size(long val) {
            size = val;
            return this;
        }

        public Builder createTime(long val) {
            createTime = val;
            return this;
        }

        public Builder done(boolean val) {
            done = val;
            return this;
        }

        public Builder path(String val) {
            path = val;
            return this;
        }

        public Builder progress(int val) {
            progress = val;
            return this;
        }

        public Builder rate(String val) {
            rate = val;
            return this;
        }

        public Builder fileTAG(int val) {
            fileTAG = val;
            return this;
        }

        public Builder md5(String val) {
            md5 = val;
            return this;
        }

        public TransLocalFile build() {
            return new TransLocalFile(this);
        }
    }
}

File: src/main/java/com/j/ming/socket/model/TransLocalFile.kt
package com.j.ming.socket.model


/**
 * Created by Sunny on 2017/5/13 0013.
 */

class TransLocalFile private constructor(builder: Builder) {
    var md5: String? = null
    val name: String?
    val size: Long
    var path: String? = ""

    var createTime: Long = 0
    var isDone = false
    var progress = 0
    var rate: String? = null
    var fileTAG = -1
    var status: Int = STATUS_UNSPECIFIED

    init {
        md5 = builder.md5
        name = builder.name
        size = builder.size
        path = builder.path
        createTime = builder.createTime
        isDone = builder.done
        progress = builder.progress
        rate = builder.rate
        fileTAG = builder.fileTAG
        status = builder.status

    }

    class Builder {
        var name: String? = null
        var size: Long = 0
        var createTime: Long = 0
        var done: Boolean = false
        var path: String? = null
        var progress: Int = 0
        var rate: String? = null
        var fileTAG: Int = 0
        var md5: String? = null
        var status = STATUS_UNSPECIFIED

        fun name(`val`: String): Builder {
            name = `val`
            return this
        }

        fun size(`val`: Long): Builder {
            size = `val`
            return this
        }

        fun createTime(`val`: Long): Builder {
            createTime = `val`
            return this
        }

        fun done(`val`: Boolean): Builder {
            done = `val`
            return this
        }

        fun path(`val`: String): Builder {
            path = `val`
            return this
        }

        fun progress(`val`: Int): Builder {
            progress = `val`
            return this
        }

        fun rate(`val`: String): Builder {
            rate = `val`
            return this
        }

        fun fileTAG(`val`: Int): Builder {
            fileTAG = `val`
            return this
        }

        fun md5(`val`: String): Builder {
            md5 = `val`
            return this
        }

        fun status(`val`: Int): Builder{
            status = `val`
            return this
        }

        fun build(): TransLocalFile {
            return TransLocalFile(this)
        }
    }

    companion object {
        const val TAG_SEND = 0
        const val TAG_RECEIVE = 1

        const val STATUS_UNSPECIFIED = -1
        const val STATUS_UPLOADING = 1
        const val STATUS_SUSPEND = 2
        const val STATUS_CANCEL = 3
        const val STATUS_FAILED = 4
        const val STATUS_SUCCEED = 5
        const val STATUS_WAITING = 6

    }
}
